### ISSUE TITLE
process full-fledged Rmd

### ISSUE THREAD

**Issue by baptiste** [jennybc/reprex#60]
I wrote an answer [for SO](http://stackoverflow.com/questions/41438206/how-to-add-arrow-of-variable-size-to-ggplot-graph/41438272#41438272) in the form of a [full Rmd document](http://rpubs.com/baptiste/arrow) for convenience, but knitr produces md not fullly compatible (code blocks, for one thing). It would be great if reprex included knitr hooks / rmarkdown custom flavoured md output format for SO. (there may well be one out there already, my memory and google haven't been very helpful). @yihui ?

**Comment by yihui** [comment]
SO does not support fenced code blocks, so you need to indent by 4 spaces, and you can use the output format `rmarkdown::md_document`.

**Comment by jennybc** [comment]
reprex does already do this, using `md_document` as @yihui points out. Request `venue = "so"`, like so:

``` r
library(reprex)
so_md <- reprex({
  x <- 1:4
  y <- 2:5
  x + y
}, venue = "so")
writeLines(so_md)
#> <!-- language-all: lang-r -->
#> <br/>
#> 
#>     x <- 1:4
#>     y <- 2:5
#>     x + y
#>     #> [1] 3 5 7 9
```

Note the output you want (SO-ready markdown) should also be waiting on your clipboard after this little exercise.

I am tickled that I just used reprex to make a reprex, formatted for GitHub, demonstrating formatting of StackOverflow. META.

**Comment by jennybc** [comment]
But, @baptiste, back to your title. reprex takes R code as input, not Rmd. I think that's the right call, because these little snippets are going to be mostly R code. You can still include prose, by prefixing with `#'`. The examples and README show this. When I use it on GitHub, I often write the prose part of my comment normally, then paste the reprex result in.

**Comment by baptiste** [comment]
Thanks, I overlooked (forgot about)  md_document. Was looking for the SO
equivalent of gfm, but that's just plain md. Yaml makes it harder to know
the available options, compared to other environments with code completion
and hints. Still, it could be nice if reprex detected rmd input, and
processed it with imgur + corresponding md output. Some people like verbose
questions or answers, with more text than code, and the spin strategy isn't
as nice in this case. That being said, I'm usually in the terse one lineR
category so it's mostly a theoretical suggestion! Thanks

**Comment by jennybc** [comment]
OK I will close for now, but keep this in mind.

I think if you've already psyched yourself up to write Rmd, a workflow helper like reprex isn't so relevant anymore. Because you've already done most of the busy work that reprex handles anyway! The other 2 things it brings to the table then are:

  * `upload.fun` defaulting to imgur upload, as @baptiste points out
  * putting Markdown on the clipboard
